appservice/src/deploy/waitForDeploymentToComplete.ts

```diff
@@ -72,35 +78,43 @@ export async function waitForDeploymentToComplete(client: SiteClient, kuduClient
-            .sort((a: DeployResult, b: DeployResult) => nonNullProp(b, 'startTime').valueOf() - nonNullProp(a, 'startTime').valueOf())
-            .shift();
-        if (deployment && !deployment.isTemp) {
-            // Make note of the id once the deplyoment has shifted to the "permanent" phase, so that we can use that to find the deployment going forward
-            permanentId = deployment.id;
-        }
-    } else {
-        // Use "latest" to get the deployment before we know the "initialReceivedTime" or "permanentId"
-        deployment = <DeployResult | undefined>await kuduClient.deployment.getResult('latest');
-        if (deployment && deployment.startTime) {
-            // Make note of the startTime because that is when kudu has began the deployment process,
-            // so that we can use that to find the deployment going forward
-            initialStartTime = deployment.startTime;
+    try {
+        if (permanentId) {
+            // Use "permanentId" to find the deployment during its "permanent" phase
+            deployment = await kuduClient.deployment.getResult(permanentId);
+        } else if (expectedId) {
+            // if we have a "expectedId" we know which deployment we are looking for, so wait until latest id reflects that
+            const latestDeployment: DeployResult = await kuduClient.deployment.getResult('latest');
+            permanentId = latestDeployment.id === expectedId ? latestDeployment.id : undefined;
+        } else if (initialStartTime) {
+            // Use "initialReceivedTime" to find the deployment during its "temp" phase
+            deployment = (await kuduClient.deployment.getDeployResults())
+                // tslint:disable-next-line:no-non-null-assertion
+                .filter((deployResult: DeployResult) => deployResult.startTime && deployResult.startTime >= initialStartTime!)
+                .sort((a: DeployResult, b: DeployResult) => nonNullProp(b, 'startTime').valueOf() - nonNullProp(a, 'startTime').valueOf())
+                .shift();
+            if (deployment && !deployment.isTemp) {
+                // Make note of the id once the deplyoment has shifted to the "permanent" phase, so that we can use that to find the deployment going forward
+                permanentId = deployment.id;
+            }
+        } else {
+            // Use "latest" to get the deployment before we know the "initialReceivedTime" or "permanentId"
+            deployment = <DeployResult | undefined>await kuduClient.deployment.getResult('latest');
+            if (deployment && deployment.startTime) {
+                // Make note of the startTime because that is when kudu has began the deployment process,
+                // so that we can use that to find the deployment going forward
+                initialStartTime = deployment.startTime;
+            }
         }
+    } catch (error) {
+        // swallow error since "latest" might not exist yet
```

Comment:
Can we minimize the scope of this try catch? Like just put it around the call to get latest? And just swallow the 404 error? Or whatever it is...